packages/pluggableWidgets/notifications-native/src/Notifications.tsx

```diff
@@ -1,67 +1,139 @@
+interface IPushNotification {
+    playSound?: boolean;
+    subText?: string;
+    userInteraction: boolean;
+    id?: number;
+    title: string;
+    message: string;
+    channelId?: string;
+    foreground: boolean;
+    data: ActionData & {
+        actionIdentifier?: string;
+        userInteraction?: number;
+    }; // iOS
+}
+
+interface ActionData {
     actionName?: string;
     guid?: string;
 }
 
-export function Notifications(props: NotificationsProps<undefined>) {
+interface Notification {
+    data: Option<ActionData>;
+    title: Option<string>;
+    body: Option<string>;
+    subTitle: Option<string>;
+}
+
+export function Notifications(props: NotificationsProps<undefined>): null {
     const listeners = useRef<Array<() => void>>([]);
     const [loadNotifications, setLoadNotifications] = useState(false);
 
-    const handleNotification = (
-        notification: FirebaseMessagingTypes.Notification,
-        data: NotificationData | undefined,
-        getHandler: (action: ActionsType) => ActionValue | undefined
-    ): void => {
-        const body: string = notification.body ?? "";
-        const title: string = notification.title ?? "";
-        const subtitle: string = notification.ios?.subtitle ?? "";
-        const actions = props.actions.filter(item => item.name === data?.actionName);
-
-        if (actions.length === 0) {
-            return;
-        }
+    const handleNotification = useCallback(
+        (notification: Notification, getHandler: (action: ActionsType) => Option<ActionValue>): void => {
+            const body: string = notification.body ?? "";
+            const title: string = notification.title ?? "";
+            const subtitle: string = notification.subTitle ?? "";
+            const actions = props.actions.filter(item => item.name === notification.data?.actionName);
 
-        if (props.guid) {
-            props.guid.setValue(data?.guid);
-        }
-        if (props.title) {
-            props.title.setValue(title);
-        }
-        if (props.subtitle) {
-            props.subtitle.setValue(subtitle);
-        }
-        if (props.body) {
-            props.body.setValue(body);
-        }
-        if (props.action) {
-            props.action.setValue(actions.join(" "));
-        }
+            if (actions.length === 0) {
+                return;
+            }
+            if (props.guid) {
+                props.guid.setValue(notification.data?.guid);
+            }
+            if (props.title) {
+                props.title.setValue(title);
+            }
+            if (props.subtitle) {
+                props.subtitle.setValue(subtitle);
+            }
+            if (props.body) {
+                props.body.setValue(body);
+            }
+            if (props.action) {
+                props.action.setValue(actions.map(action => action.name).join(" "));
+            }
```

Comment:
These could all use the `?.`, right?

Reply:
Do you mean instead of guarding them?

Reply:
Yeah, `props.guid?.setValue` instead of `if (props.guid) { ...` :)

Reply:
I will change, it reduces the original code a lot. Nice!

Reply:
Done